modules/afsocket/afinet-dest.c

```diff
@@ -118,6 +118,43 @@ afinet_dd_set_tls_context(LogDriver *s, TLSContext *tls_context)
                                         afinet_dd_verify_callback, self);
 }
 
+void
+afinet_dd_add_failovers(LogDriver *s, GList *failovers)
+{
+  AFInetDestDriver *self = (AFInetDestDriver *)s;
+  if (!self->server_candidates)
+    {
+      self->server_candidates = g_list_append(self->server_candidates, self->hostname);
+    }
+  while (failovers)
+    {
+      self->server_candidates = g_list_append(self->server_candidates, g_strdup(failovers->data));
+      g_free(failovers->data);
+      failovers = g_list_delete_link(failovers, failovers);
+    }
+  self->server_candidates = g_list_first(self->server_candidates);
+  self->hostname = self->server_candidates->data;
+}
+
+static gchar *
+_get_next_destination_candidate(AFInetDestDriver *self)
+{
+  gchar *candidate = self->hostname;
+  if (self->server_candidates)
+    {
+      if (self->server_candidates->next)
+        {
+          self->server_candidates = self->server_candidates->next;
+        }
+      else
+        {
+          self->server_candidates = g_list_first(self->server_candidates);
+        }
+      candidate = (gchar *)self->server_candidates->data;
+    }
```

Comment:
I would probably not mutate the server_candidate list to track the current server, why don't we have a separate current_server_candidate pointer pointing the current element. I think that would make things simpler to read.